### PR TITLE
LX-1826 /etc/issue file created/populated after not-in-place upgrade

### PR DESCRIPTION
**Pull request by pandar00**
As noted in the bug, we are seeing a default `/etc/issue` file present on VM upgraded with "not-in-place" upgrade.

**Cause**
* The current implementation of `migrate_file` function only migrates the file to the container if it exists on the host. This means if a host doesn't have the file but the container somehow does, then it'll leave the file in the container.
This case happens with `/etc/issue` file where host doesn't have the file, but the `not-in-place` container created it while executing`debootstrap` which has the content "Ubuntu 18.04.2 LTS \n \l".

**Fix**
* We force delete the migration target file on the container file system if the file doesn't exist on the host.

### CODE REVIEW

**Comment by pandar00** [comment]
@dtzeng @prakashsurya 
I'll continue testing if you guys agree with the fix

**Comment by prakashsurya** [comment]
I'm not sure I agree with this approach in general, but let me think about it a bit more.

For the `/etc/issue` file specifically, we remove it during the initial build, via an ansible role; see [here](https://github.com/delphix/appliance-build/blob/master/live-build/misc/ansible-roles/appliance-build.minimal-common/tasks/main.yml#L53).

So, I wouldn't be opposed to something more targeted to this file in particular.. but maybe what you propose is actually best, in general.. i.e. if the file doesn't exist pre-upgrade, perhaps it shouldn't exist post-upgrade either. Hm..

**Comment by pandar00** [comment]
> So, I wouldn't be opposed to something more targeted to this file in particular.. 

I'm fine with this approach too

I'm also not in any hurry to fix the bug so take time as needed. I'll also try to see if there are better options.

File: upgrade/upgrade-scripts/upgrade-container [hunk 1]
```diff
@@ -428,6 +428,12 @@ function migrate_file() {
 			die "'mkdir -p $parentdir' failed for '$CONTAINER'"
 		cp "$path" "/var/lib/machines/${CONTAINER}${path}" ||
 			die "'cp $path' failed for '$CONTAINER'"
+	else
+		# If a migration target file doesn't exist pre-upgrade then
```
Comment (prakashsurya) [inline upgrade/upgrade-scripts/upgrade-container]: Can we reword this to something like: "If the target file doesn't exist" .. it's minor, but since we use "migration" as the name for illumos to linux upgrades, it can be a little confusing to call the file a "migration target file" (e.g. if one assumes "migration" used here means "illumos to linux migration").

File: upgrade/upgrade-scripts/upgrade-container [hunk 2]
```diff
@@ -428,6 +428,12 @@ function migrate_file() {
...
+		# we delete the file. This case could happen if pre-upgrade
+		# configuration doesn't exist but a refresh install placed
```
Comment (prakashsurya) [inline upgrade/upgrade-scripts/upgrade-container]: Should "refresh" be "fresh" ?

File: upgrade/upgrade-scripts/upgrade-container [hunk 3]
```diff
@@ -428,6 +428,12 @@ function migrate_file() {
...
+		# package default configuration.
+		rm -f "/var/lib/machines/${CONTAINER}${path}"
```
Comment (prakashsurya) [inline upgrade/upgrade-scripts/upgrade-container]: add `|| die "failed to remove '$path' in '$CONTAINER'"` in case `rm` fails? since we're using `-f`, even if the file doesn't exist in the container, it shouldn't report an error.

**Comment by pandar00** [comment]
`git-ab-pre-push` is [here](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/874/)
